conan/tools/cmake/toolchain/blocks.py

```diff
@@ -486,7 +486,9 @@ def context(self):
         for req in host_req:
             cppinfo = req.cpp_info.aggregated_components()
             # If the builddir is the package_folder, then it is the default "root" one
-            nf = os.path.normpath(req.package_folder)
+            # The package_folder can be None if editable and layout(), in that case only the
+            # host_build_paths_noroot will be populated
+            nf = os.path.normpath(req.package_folder or "")
             host_build_paths_root.extend(p for p in cppinfo.builddirs if os.path.normpath(p) == nf)
             host_build_paths_noroot.extend(p for p in cppinfo.builddirs if os.path.normpath(p) != nf)
```

Comment:
I would see this more clear:

```suggestion
            if req.package_folder:
                nf = os.path.normpath(req.package_folder)
                host_build_paths_root.extend(p for p in cppinfo.builddirs if os.path.normpath(p) == nf)
                host_build_paths_noroot.extend(p for p in cppinfo.builddirs if os.path.normpath(p) != nf)
            else:
                host_build_paths_root = []
                host_build_paths_noroot.extend(p for p in cppinfo.builddirs)
```